src/go/rpk/pkg/api/admin/admin.go

```diff
@@ -185,39 +215,112 @@ func (a *AdminAPI) GetLeaderID() (*int, error) {
 
 // sendAny sends a single request to one of the client's urls and unmarshals
 // the body into into, which is expected to be a pointer to a struct.
+//
+// On errors, this function will keep trying all the nodes we know about until
+// one of them succeeds, or we run out of nodes.  In the latter case, we will return
+// the error from the last node we tried.
 func (a *AdminAPI) sendAny(method, path string, body, into interface{}) error {
-	pick := rng(len(a.urls))
-	url := a.urls[pick] + path
-	res, err := a.sendAndReceive(context.Background(), method, url, body)
-	if err != nil {
-		return err
+	// Shuffle the list of URLs
+	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	shuffled := make([]string, len(a.urls))
+	copy(shuffled, a.urls)
+	rng.Shuffle(len(shuffled), func(i, j int) { shuffled[i], shuffled[j] = shuffled[j], shuffled[i] })
+
+	var err error
+	for i := range shuffled {
+		url := shuffled[i] + path
+		var res *http.Response
+
+		// Where there are multiple nodes, disable the HTTP request retry in favour of our
+		// own retry across the available nodes
```

Comment:
Am I correct in reading this:  for `sendAny` requests where the user's rpk configuration only specifies one admin URL, this will use the retryClient, which will retry up to three times, with 1.5s+jitter(100ms) backoff each time?

Is it worth it to reduce the retry timeout and increase the number of retries? 1.5s per attempt seems a bit slow.

Reply:
The intent here is:
- for a single admin API endpoint, if we can't reach it then we presume something relatively slow-moving is the problem (e.g. a leader election that can take >=1.5s, or a node being offline).  The system is in a degraded state, so we don't want to add to that by retrying really aggressively.  If we retried much more frequently (e.g. 100ms) then the user would probably still be waiting a second or two because that's just how long the election takes -- they only benefit from a fast retry if they're lucky enough to be attempting the operation right before the election was going to complete anyway.
- for multiple endpoints, we do not wait before retrying on the next node, because user might leave a single node offline for extended periods, and we should endeavor to still be fairly snappy when 2/3 nodes are down.

Reply:
It still may be worth it to use ever so slightly more aggressive retries, but I think we can wait for user experience reports here. I think more likely is a user will be talking to a non-local broker on an unreliable internet connection -- if they only configured one broker in their local redpanda.yaml, then they may get a slower experience (although, if the connection is unreliable, maybe 1.5s between retries is expected anyway).